File: Lista_Logica_Programacao/Java/Ex9.java
// Exercício 9 - Calculadora Simples com Métodos

import java.util.Scanner;

public class Ex9 {
    // métodos estáticos para cada operação
    // cada método recebe dois números e retorna o resultado

    public static double somar(double a, double b) {
        return a + b;
    }

    public static double subtrair(double a, double b) {
        return a - b;
    }

    public static double multiplicar(double a, double b) {
        return a * b;
    }

    // método dividir com if/else para cuidar de divisão por zero
    public static Double dividir(double a, double b) {
        if (b == 0) {
            return null;  // retorna null se não conseguir dividir
        } else {
            return a / b;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.println("\n=== Calculadora ===");
            System.out.println("1 - Soma");
            System.out.println("2 - Subtração");
            System.out.println("3 - Multiplicação");
            System.out.println("4 - Divisão");
            System.out.println("5 - Sair");

            System.out.print("Escolha a operação (1-5): ");
            int escolha = sc.nextInt();

            if (escolha == 5) {
                System.out.println("Saindo...");
                break;
            }

            System.out.print("Digite o primeiro número: ");
            double num1 = sc.nextDouble();
            System.out.print("Digite o segundo número: ");
            double num2 = sc.nextDouble();

            // verifica qual operação fazer e chama o método apropriado
            if (escolha == 1) {
                double resultado = somar(num1, num2);  // chama método somar
                System.out.println("Resultado da soma: " + resultado);
            } else if (escolha == 2) {
                double resultado = subtrair(num1, num2);  // chama método subtrair
                System.out.println("Resultado da subtração: " + resultado);
            } else if (escolha == 3) {
                double resultado = multiplicar(num1, num2);  // chama método multiplicar
                System.out.println("Resultado da multiplicação: " + resultado);
            } else if (escolha == 4) {
                Double resultado = dividir(num1, num2);  // chama método dividir
                if (resultado == null) {
                    System.out.println("Erro: divisão por zero!");
                } else {
                    System.out.println("Resultado da divisão: " + resultado);
                }
            } else {
                System.out.println("Opção inválida, tente novamente.");
            }
        }

        sc.close();
    }
}
/*
Anotações:

- public static é necessário para criar métodos que a main pode chamar.
- return devolve o resultado do método quando ele é chamado.
- cada método (somar, subtrair, multiplicar, dividir) faz uma operação.
- Double (com D maiúsculo) permite retornar null para erro na divisão.
- if/elif/else escolhem qual método chamar conforme a opção.
- métodos deixam o código mais organizado e reutilizável.
- null significa "sem valor", usado para indicar erro na divisão.
*/

File: Lista_Logica_Programacao/Python/ex9.py
# Exercício 9 - Calculadora Simples

# programa que mostra um menu e realiza operações básicas
# o usuário escolhe a operação, insere dois números e o resultado é exibido

while True:
    # mostra o menu de opções
    print("\n=== Calculadora ===")
    print("1 - Soma")
    print("2 - Subtração")
    print("3 - Multiplicação")
    print("4 - Divisão")
    print("5 - Sair")

    escolha = input("Escolha a operação (1-5): ")

    if escolha == "5":
        print("Saindo...")
        break  # encerra o loop

    # pede os números; converto para float para aceitar décimais
    num1 = float(input("Digite o primeiro número: "))
    num2 = float(input("Digite o segundo número: "))

    # verifica qual operação fazer
    if escolha == "1":
        resultado = num1 + num2
        print(f"Resultado da soma: {resultado}")
    elif escolha == "2":
        resultado = num1 - num2
        print(f"Resultado da subtração: {resultado}")
    elif escolha == "3":
        resultado = num1 * num2
        print(f"Resultado da multiplicação: {resultado}")
    elif escolha == "4":
        # cuidado com divisão por zero
        if num2 == 0:
            print("Erro: divisão por zero!")
        else:
            resultado = num1 / num2
            print(f"Resultado da divisão: {resultado}")
    else:
        print("Opção inválida, tente novamente.")

# Anotações:
# - usei while True para que o menu volte até o usuário escolher sair.
# - input() retorna string, por isso comparo escolha com "1", "2" etc.
# - converti os números para float porque podem ser com ponto.
# - if/elif/else verificam a operação escolhida.
# - cuidei da divisão por zero para não gerar erro.
# - coloquei comentários e notas como nos outros exercícios, explicando cada parte para revisar com o professor.